sat-plugin/src/main/resources/rulesets/checkstyle/rules.xml

```diff
@@ -40,39 +41,49 @@
     <property name="header"
         value="${checkstyle.headerCheck.content}"
         default="^/\*\*$\n^ \* Copyright \(c\) {0}-{1} by the respective copyright holders\.$\n^ \*$\n^ \* All rights reserved\. This program and the accompanying materials$\n^ \* are made available under the terms of the Eclipse Public License v1\.0$\n^ \* which accompanies this distribution, and is available at$\n^ \* http://www.eclipse.org/legal/epl\-v10\.html$"/>
+    <type value="java"/>
   </module>
   
   <module name="NewlineAtEndOfFile">
     <property name="severity" value="info"/>
+    <type value="java"/>
   </module>
 
   <module name="org.openhab.tools.analysis.checkstyle.ExportInternalPackageCheck">
     <property name="severity" value="error" />
+    <type value="manifest"/>
   </module>
     
   <module name="org.openhab.tools.analysis.checkstyle.ManifestExternalLibrariesCheck">
     <property name="severity" value="error" />
+    <type value="manifest"/>
   </module>
   
   <module name="org.openhab.tools.analysis.checkstyle.BuildPropertiesExternalLibrariesCheck">
     <property name="severity" value="error" />
+    <type value="properties"/>
   </module>
   
   <module name="org.openhab.tools.analysis.checkstyle.OutsideOfLibExternalLibrariesCheck">
     <property name="severity" value="error" />
     <property name="ignoredDirectories" value="${checkstyle.outsideOfLibExternalLibrariesCheck.ignoredDirectories}" default="target,bin" />
+    <type value="properties"/>
   </module>
   
   <module name="org.openhab.tools.analysis.checkstyle.readme.MarkdownCheck">
     <property name="severity" value="warning" />
+    <type value="properties"/>
+    <type value="markdown"/>
```

Comment:
What is the consequence if one of the two types is requested to be skipped, should we document this behaviour?

Reply:
If one of the two types is requested to be skipped the check will be skipped. I added this to the docs.